llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1778,10 +1778,19 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
       break;
     }
 
-    if (IndexOp.getImm() != 0 &&
-        SrcTy.getElementCount().getKnownMinValue() % IndexOp.getImm() != 0) {
-      report("Index must be a multiple of the source vector's minimum vector "
-             "length",
+    int64_t Idx = IndexOp.getImm();
+    auto DstMinLen = DstTy.getElementCount().getKnownMinValue();
```

Comment:
I'm not sure `auto` here is consistent with llvm's guidelines on using `auto` does getKnownMinValue() return `unsigned` or `uint64_t`? I honestly don't know.